src/main/java/de/adesso/blogpostchecker/CheckExecutor.java

```diff
@@ -0,0 +1,97 @@
+package de.adesso.blogpostchecker;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CheckExecutor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CheckExecutor.class);
+
+    public void executeChecks(PostHeader postHeader, String authors) {
+        checkCategoryOnlyOneEntry(postHeader);
+        checkTagsNotEmpty(postHeader);
+        checkAuthorNotEmpty(postHeader);
+        checkAuthorInYml(postHeader, authors);
+        checkTitleNotEmpty(postHeader);
+        checkLayoutCorrect(postHeader);
+        checkDateMatchesFormat(postHeader);
+    }
+
+    private void checkCategoryOnlyOneEntry(PostHeader header) {
+        if (header.getCategories() != null && !header.getCategories().equals("")) {
+            if (!header.getCategories().contains(",")) {
```

Comment:
Was ist mit etwas wie  
```java
header.getCategories() != null && header.size() == 1
```
? Somit könnten wir die doppelte if Abfrage vermeiden.

Reply:
An der Stelle werden zwei Aspekte überprüft. Zum Einen, dass die Category nicht leer ist, zum Anderen, dass nur eine Category angegeben ist (Kein Komma vorhanden). Wir können es aber gut in zwei unterschiedliche Methoden auslagern. `size() == 1` würde nicht funktionieren, da damit der entsprechende Test nicht realisiert ist. Aber das mit zwei Methoden ist eine gute Idee.